### DISCUSSION TITLE
Transportation problem

### DISCUSSION THREAD

**Discussion by caumente** [coin-or/pulp#496] [Q&A]
Hi all,

First of all, I would like to thank @pchtsp and all people who work on developing PuLP. Great work!

I've been solving the deterministic transportation problem using PuLP. However, I wonder if the library provides any technique to add uncertainty to the process.

Let's imagine that the demand of the customers is based on a given distribution instead of knowing the precise value. Is there any function that allows doing this?

Thanks a million!

**Comment by pchtsp** [comment]
PuLP itself does not help in the automatic migration of deterministic problems into stochastic optimization problems. If you reformulate your problem into its "stochastic equivalent" you can then use PuLP to code it and solve it. 

Reformulation usually requires including new sets (scenarios) to the input data and then choosing carefully when is each variable decided (before new information is available or after). If you read some resources on that you can actually use PuLP to generate the stochastic optimization model. Unfortunately, I do not have any resources on that at hand. That said, there's definitively many examples and resources that cover stochastic optimization modelling.

I think the pyomo guys have some stochastic optimization functionality (https://github.com/Pyomo/pysp) but I have never used it so I can't tell you much more.

**Reply by caumente** [reply]
Hi @pchtsp, thanks for your quick answer. I will definitely turn the stochastic problem into a deterministic problem. It seems that is the easiest approach when it is possible to reformulate the stochastic constraints. 

Thanks also for the resource Pyomo ;)

**Comment by pchtsp** [comment]
The elastic constraints don't have anything to do with the stochastic
formulation of a decision problem, unfortunately.

This is one ppt I just found that explains some ways to formulate the
stochastic optimization problem linearly:

http://cermics.enpc.fr/~delara/exposes/slides_introduction_stochastic_optimization.pdf

**Reply by caumente** [reply]
Hi @pchtsp, thanks for sharing that resource.

That is exactly what I will do. To reformulate the problem to describe it as a deterministic problem.
